urllib3/_collections.py

```diff
@@ -304,17 +304,17 @@ def items(self):
         return list(self.iteritems())
 
     @classmethod
-    def from_httplib(cls, message, duplicates=('set-cookie',)): # Python 2
+    def from_httplib(cls, message): # Python 2
         """Read headers from a Python 2 httplib message object."""
-        ret = cls(message.items())
-        # ret now contains only the last header line for each duplicate.
-        # Importing with all duplicates would be nice, but this would
-        # mean to repeat most of the raw parsing already done, when the
-        # message object was created. Extracting only the headers of interest 
-        # separately, the cookies, should be faster and requires less
-        # extra code.
-        for key in duplicates:
-            ret.discard(key)
-            for val in message.getheaders(key):
-                ret.add(key, val)
-            return ret
+        headers = []
+         
+        for line in message.headers:
+            if line.startswith((' ', '\t')):
```

Comment:
My previous implementation sucked, as it only copied all single header items, and then added the explicitly requested multiheaders. This obviously left out all not explicitly multiheaders asked for and reduced them to only the last item. See my previous comment.

The new code is re-iterating the raw header information completely, including any multiheader. The code is taken from @wtolson, link see above, thanks for that again! I also extended the header parsing testcase for checking multiple lines. Might be worth, to borrow some further tests from original python2.7 headers ...